sdks/go/pkg/beam/testing/passert/floats.go

```diff
@@ -26,6 +26,56 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+// EqualsFloat checks that two PCollections of floats are equal, with each element
+// being within a specified threshold of its corresponding element. Both PCollections
+// are loaded into memory, sorted, and compared element by element.
+func EqualsFloat(s beam.Scope, observed, expected beam.PCollection, threshold float64) {
+	s = s.Scope(fmt.Sprintf("passert.EqualsFloat[%v]", threshold))
+	beam.ParDo0(s, &thresholdFn{threshold: threshold}, beam.Impulse(s), beam.SideInput{Input: observed}, beam.SideInput{Input: expected})
+}
+
+type thresholdFn struct {
+	threshold float64
```

Comment:
Export your Field names please.

While passert functions are intended to be run on single machines for testing, unexported Fields are never serialized as part of the DoFn, which makes them not function as intended on Portable runners.

Basically, all the rules the compiler enforces about Exported and unexported names, are respected by the underlying "reflect" package as well for access. This means that unexported fields can't be read and then subsequently written to reflectively, effectively dropping their data entirely.

It's legal to have DoFns that only have unexported fields (though it probably shouldn't be), and we likely should add a validation that there's at least one exported field for types used as PCollection elements. It's my number one comment to users with issues...

(unexported type and function names are fine though)

Reply:
Done